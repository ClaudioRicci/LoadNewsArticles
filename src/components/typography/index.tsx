import styled from "styled-components";
import { colors, fonts, align, deviceSizes } from "../../themeVariables";

export const H1Tag = styled.h1`
  font: 900 1em/1rem ${fonts.header};
  color: ${colors.primary};
  text-align: ${align.left};
  border-bottom: 0.2rem ${colors.primary} solid;
  padding-bottom: 0.2rem;
  width: fit-content;
  grid-area: a;
  @media (max-width: ${deviceSizes.mobile}) {
    font: 900 1.4em/1.4rem ${fonts.header};
  }
`;

export const H2Tag = styled.h2`
  font: 900 0.8em/0.8rem ${fonts.header};
  color: ${colors.secondary};
  line-height: 1rem;
`;

export const ArticleTitle = styled.a`
  font: 900 0.84em/0.84rem ${fonts.header};
  color: ${colors.secondary};
  text-align: ${align.left};
  text-decoration: none;
  margin-bottom: 0.5rem;
  cursor: pointer;
`;

export const PTag = styled.p`
  line-height: 1rem;
`;
