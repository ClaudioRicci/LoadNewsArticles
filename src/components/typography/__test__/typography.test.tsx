import React from "react";
import ReactDOM from "react-dom";
import { H1Tag, H2Tag, ArticleTitle, PTag } from "../../typography";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

afterEach(cleanup);

describe("Typography Tests", () => {
  describe("H1Tag Tests", () => {
    it("H1 Tag renders correctly", () => {
      const div = document.createElement("div");
      ReactDOM.render(<H1Tag />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    test("H1 is primary colour", () => {
      const tree = renderer.create(<H1Tag />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule("color", "#D9819D");
    });
  });

  describe("H2Tag Tests", () => {
    it("H2Tag renders correctly", () => {
      const div = document.createElement("div");
      ReactDOM.render(<H2Tag />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    test("H1 is secondary colour", () => {
      const tree = renderer.create(<H2Tag />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule("color", "#5D5D5D");
    });
  });

  describe("ArticleTitle Tests", () => {
    it("ArticleTitle renders correctly", () => {
      const div = document.createElement("div");
      ReactDOM.render(<ArticleTitle />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    test("ArticleTitle is secondary colour", () => {
      const tree = renderer.create(<ArticleTitle />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule("color", "#5D5D5D");
    });

    test("ArticleTitle has a cursor pointer to denote a link", () => {
      const tree = renderer.create(<ArticleTitle />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule("cursor", "pointer");
    });
  });

  describe("PTag Tests", () => {
    it("PTag renders correctly", () => {
      const div = document.createElement("div");
      ReactDOM.render(<PTag />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    test("PTag has 1rem line-height", () => {
      const tree = renderer.create(<PTag />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule("line-height", "1rem");
    });
  });
});
