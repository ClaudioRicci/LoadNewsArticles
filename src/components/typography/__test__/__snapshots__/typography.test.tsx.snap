// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Tests ArticleTitle Tests ArticleTitle has a cursor pointer to denote a link 1`] = `
.c0 {
  font: 900 0.84em/0.84rem Verdana,Arial,Helvetica,sans-serif;
  color: #5D5D5D;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 0.5rem;
  cursor: pointer;
}

<a
  className="c0"
/>
`;

exports[`Typography Tests ArticleTitle Tests ArticleTitle is secondary colour 1`] = `
.c0 {
  font: 900 0.84em/0.84rem Verdana,Arial,Helvetica,sans-serif;
  color: #5D5D5D;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 0.5rem;
  cursor: pointer;
}

<a
  className="c0"
/>
`;

exports[`Typography Tests H1Tag Tests H1 is primary colour 1`] = `
.c0 {
  font: 900 1em/1rem Verdana,Arial,Helvetica,sans-serif;
  color: #D9819D;
  text-align: left;
  border-bottom: 0.2rem #D9819D solid;
  padding-bottom: 0.2rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  grid-area: a;
}

@media (max-width:768px) {
  .c0 {
    font: 900 1.4em/1.4rem Verdana,Arial,Helvetica,sans-serif;
  }
}

<h1
  className="c0"
/>
`;

exports[`Typography Tests H2Tag Tests H1 is secondary colour 1`] = `
.c0 {
  font: 900 0.8em/0.8rem Verdana,Arial,Helvetica,sans-serif;
  color: #5D5D5D;
  line-height: 1rem;
}

<h2
  className="c0"
/>
`;

exports[`Typography Tests PTag Tests PTag has 1rem line-height 1`] = `
.c0 {
  line-height: 1rem;
}

<p
  className="c0"
/>
`;
